plugins/filestorage/localfs/index.js

```diff
@@ -290,6 +291,57 @@ LocalFileStorage.prototype.getFileStats = function (filePath, callback) {
 };
 
 /**
+ * Copies an asset from one tenant to another.
+ * 
+ * @param {object} asset - A valid 'asset' record
+ * @param {string} sourceTenantName - Source tenant.name
+ * @param {string} destinationTenantName - Destination tenant.name
+ */
+LocalFileStorage.prototype.copyAsset = function(asset, sourceTenantName, destinationTenantName, callback) {
+  // Verify the destination folder exists
+  var assetFolder = path.join(configuration.serverRoot, configuration.getConfig('dataRoot'), destinationTenantName, asset.directory);
+
+  mkdirp(assetFolder, function(error) {
+    if (error) {
+      logger.log('error', 'mkdirp error while copying asset, creating ' + assetFolder);
+      return callback(error);
+    }
+
+    // Copy the asset
+    var sourceFile = path.join(configuration.serverRoot, configuration.getConfig('dataRoot'), sourceTenantName, asset.path);
+    var destinationFile = path.join(configuration.serverRoot, configuration.getConfig('dataRoot'), destinationTenantName, asset.path);
+
+    ncp(sourceFile, destinationFile, {clobber:false}, function(error) {
+      if (error) {
+        logger.log('error', 'ncp error while copying ' + asset.filename);
```

Comment:
should we also log the error message here (or is it logged elsewhere)? it could be file exists, or a file permissions problem or something else like disk space. I know from painful experience that “couldn't save $filename" can be a frustrating error message unless you know why.